metals/src/main/scala/scala/meta/metals/compiler/ScalacProvider.scala

```diff
@@ -35,26 +35,28 @@ class ScalacProvider()(implicit client: JsonRpcClient) extends LazyLogging {
       // Looking up by sourceDirectory alone is not enough since
       // in sbt it's possible to `sources += file("blah")`, which would
       // not be respected if we only went by directories.
-      .orElse(compilerBySourceDirectory(uri))
+      .orElse(compilerBySourceDirectory(uri).map(_._2))
```

Comment:
I saw you and @gabro discussing earlier but don’t remember the result.  Do we have a convention to avoid ._N methods?  I personally don’t like the readability here as it makes it difficult to see that we are falling back to Global.

Reply:
Good point, I agree `_N` methods are hard to read. Updated

Reply:
this might actually be a decent linter rule "Introduce a case class or use pattern matching instead! Overuse of tuples is a code smell"